specifyweb/frontend/js_src/lib/components/WbActions/WbActions.tsx

```diff
@@ -169,74 +131,120 @@ export function WbActionsComponent({
     },
   };
 
-  React.useEffect(() => {
-    if (
-      !isUploaded &&
-      (mappings?.lines ?? []).length === 0 &&
-      hasPermission('/workbench/dataset', 'upload')
-    ) {
-      openNoUploadPlan();
-    }
-  }, []);
+  return messages[mode];
+}
+
+export function WbActions({
+  dataset,
+  hasUnsavedChanges,
+  isUploaded,
+  isResultsOpen,
+  workbench,
+  mappings,
+  checkDeletedFail,
+  onDatasetRefresh: handleRefresh,
+  onSpreadsheetUpToDate: handleSpreadsheetUpToDate,
+  onToggleResults,
+}: {
+  readonly dataset: Dataset;
+  readonly hasUnsavedChanges: boolean;
+  readonly isUploaded: boolean;
+  readonly isResultsOpen: boolean;
+  readonly workbench: Workbench;
+  readonly mappings: WbMapping | undefined;
+  readonly checkDeletedFail: (statusCode: number) => void;
+  readonly onDatasetRefresh: () => void;
+  readonly onSpreadsheetUpToDate: () => void;
+  readonly onToggleResults: () => void;
+}): JSX.Element {
+  const [canLiveValidate] = userPreferences.use(
+    'workBench',
+    'general',
+    'liveValidation'
+  );
+  const [noUploadPlan, openNoUploadPlan, closeNoUploadPlan] = useBooleanState();
```

Comment:
this should be moved into WbNoUploadPlan

that's 3 props fewer for WbNoUploadPlan

in general, try to move the state as down as you can

Reply:
ah, sorry I overlooked that